src/index.rs

```diff
@@ -332,72 +303,82 @@ impl ResourceIndex {
                     path.display()
                 );
             }
-
             self.insert_entry(path.clone(), entry.clone());
         }
 
-        let added: HashMap<CanonicalPathBuf, ResourceId> = added
+        let added: HashMap<PathBuf, ResourceId> = added
             .into_iter()
             .map(|(path, entry)| (path, entry.id))
             .collect();
 
         Ok(IndexUpdate { deleted, added })
     }
 
-    // the caller must ensure that:
-    // * the index is up-to-date except this single path
-    // * the path hasn't been indexed before
+    /// Indexes a new entry identified by the provided path, updating the index
+    /// accordingly.
+    ///
+    /// The caller must ensure that:
+    /// - The index is up-to-date except for this single path
+    /// - The path hasn't been indexed before
+    ///
+    /// Returns an error if:
+    /// - The path does not exist
+    /// - Metadata retrieval fails
     pub fn index_new(&mut self, path: &dyn AsRef<Path>) -> Result<IndexUpdate> {
-        log::debug!("Indexing a new path");
```

Comment:
Let's use name `track_addition` if it's same function as here https://github.com/ARK-Builders/arklib/blob/index/update-one/src/index.rs#L359